content/blog/2022-12-18-please-test-black-23.1a1.md

```diff
@@ -0,0 +1,616 @@
+
+```python3
+async def main():
+    await asyncio.sleep(1)
+    await (set_of_tasks | other_set)
+```
+
+#### Remove redundant parentheses in with statements
+
+**Source:**
+
+```python3
+with (open("bla.txt")):
+    pass
+
+with (open("bla.txt")), (open("bla.txt")):
+    pass
+
+with (open("bla.txt") as f):
+    pass
+
+with (open("bla.txt")) as f:
+    pass
+```
+
+**Before:** unchanged.
+
+**After** ([22.6.0+][#2926]):
+
+```python3
+with open("bla.txt"):
+    pass
+
+with open("bla.txt"), open("bla.txt"):
+    pass
+
+with open("bla.txt") as f:
+    pass
+
+with open("bla.txt") as f:
+    pass
+```
+
+#### Remove redundant parentheses from except clauses
+
+**Source:**
+
+```python3
+try:
+    a.something
+except (AttributeError) as err:
+    raise err
+```
+
+**Before:** unchanged.
+
+**After** ([22.3.0+][#2939]):
+
+```python3
+try:
+    a.something
+except AttributeError as err:
+    raise err
+```
+
+#### Remove redundant (outermost) parentheses in for statements
+
+**Source:**
+
+```python3
+for (k, v) in d.items():
+    print(k, v)
+```
+
+**Before:** unchanged.
+
+**After** ([22.3.0+][#2945]):
+
+```python3
+for k, v in d.items():
+    print(k, v)
+```
+
+Unfortunately, nested parentheses are still left untouched.
+[I have an open PR to fix that][#3243], but it isn't ready yet.
+
+### Remove blank lines after code block open
+
+<!-- 9 -->
+
+**Note**: This new feature will be applied to **all code blocks**: `def`, `class`, `if`,
+`for` , `while`, `with`, `case` and `match`.
+
+**Source:**
+
+```python3
+def foo():
+
+
+
+    print("All the newlines above me should be deleted!")
+
+
+if condition:
+
+    print("No newline above me!")
+
+    print("There is a newline above me, and that's OK!")
+
+
+class Point:
+
+    x: int
+    y: int
+
+    def as_tuple(self) -> tuple[int, int]:
+        return (self.x, self.y)
+```
+
+**Before:**
+
+```python3
+def foo():
+
+    print("All the newlines above me should be deleted!")
+
+
+if condition:
+
+    print("No newline above me!")
+
+    print("There is a newline above me, and that's OK!")
+
+
+class Point:
+
+    x: int
+    y: int
+
+    def as_tuple(self) -> tuple[int, int]:
+        return (self.x, self.y)
+```
+
+**After** ([22.6.0+][#3035]):
+
+```python3
+def foo():
+    print("All the newlines above me should be deleted!")
+
+
+if condition:
+    print("No newline above me!")
+
+    print("There is a newline above me, and that's OK!")
+
+
+class Point:
+    x: int
+    y: int
+
+    def as_tuple(self) -> tuple[int, int]:
+        return (self.x, self.y)
+```
+
+### Ignore magic trailing comma for single-element subscripts
+
+<!-- 13 -->
+
+Black exempts single-element tuple literals from the usual handling of magic trailing
+commas: `(1,)` will not have a newline added (as would happen for lists, sets, etc.).
+
+However, if you wanted to write `tuple[int,]` (to make it more visually distinctive to
+`list[int]`) Black would explode it which doesn't look great.
```

Comment:
```suggestion
However, if you wrote `tuple[int,]` (to make it more visually distinctive to
`list[int]`), Black would explode it, which doesn't look great.
```
Black can't do anything if you just wanted to write it, only if you actually write it :)